syz-verifier/verifier.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2021 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package main
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"log"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/google/syzkaller/pkg/mgrconfig"
+	"github.com/google/syzkaller/pkg/osutil"
+	"github.com/google/syzkaller/prog"
+	"github.com/google/syzkaller/vm"
+
+	"github.com/google/syzkaller/sys/targets"
+)
+
+type cfgFlagVals []string
+
+func (cfvs *cfgFlagVals) String() string {
+	return fmt.Sprint(*cfvs)
+}
+
+func (cfvs *cfgFlagVals) Set(value string) error {
+	if len(*cfvs) > 0 {
+		return errors.New("configs flag were already set")
+	}
+	for _, cfv := range strings.Split(value, ",") {
+		*cfvs = append(*cfvs, cfv)
+	}
+	return nil
+}
+
+var (
+	cfvs      cfgFlagVals
+	flagDebug = flag.Bool("debug", false, "dump all VM output to console")
+	workdir   string
+)
+
+// Verifier TODO
+type Verifier struct {
+	cfgs       []*mgrconfig.Config
+	vmPools    []*vm.Pool
+	targets    []*prog.Target
+	sysTargets []*targets.Target
```

Comment:
Since there are 4 parallel arrays already, I think it makes sense to create a struct that will hold all per-kernel info (config, pool, etc). It will be much handier to be able to pass that struct to functions, rather than passing Verifier an an index.

Reply:
Good idea, thanks! :)